packages/devtools_app/lib/src/debugger/flutter/debugger_screen.dart

```diff
@@ -88,60 +74,52 @@ class DebuggerScreenBodyState extends State<DebuggerScreenBody>
     final newController = Controllers.of(context).debugger;
     if (newController == controller) return;
     controller = newController;
-    addAutoDisposeListener(controller.isPaused, _onPaused);
-    addAutoDisposeListener(controller.breakpoints);
-  }
 
-  Future<void> _onPaused() async {
-    // TODO(https://github.com/flutter/devtools/issues/1648): Allow choice of
-    // the scripts on the stack.
-    if (controller.isPaused.value) {
-      final currentStack = await controller.getStack();
+    addAutoDisposeListener(controller.currentScript, () {
       setState(() {
-        stack = currentStack;
+        script = controller.currentScript.value;
       });
-      if (stack == null || stack.frames.isEmpty) return;
-      final currentScript =
-          await controller.getScript(stack.frames.first.location.script);
+    });
+    addAutoDisposeListener(controller.currentStack, () {
       setState(() {
-        script = currentScript;
+        stack = controller.currentStack.value;
       });
-    } else {
+    });
+    addAutoDisposeListener(controller.scriptList, () {
       setState(() {
-        stack = null;
+        scriptList = controller.scriptList.value;
       });
-    }
+    });
+    breakpoints = controller.breakpoints.value;
```

Comment:
is there a reason we can't just use the breakpoints valuelistenable from the controller?
Same question for the scriptList.
It would be nice to not have two copies of the data.